src/pages/RFOX/hooks/useTimeInPoolQuery.ts

```diff
@@ -1,37 +1,40 @@
-import { skipToken } from '@tanstack/react-query'
+import { skipToken, useQuery } from '@tanstack/react-query'
 import { foxStakingV1Abi } from 'contracts/abis/FoxStakingV1'
 import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
 import dayjs from 'dayjs'
 import { useMemo } from 'react'
 import { getAbiItem, getAddress } from 'viem'
 import { arbitrum } from 'viem/chains'
-import { useQuery } from 'wagmi/query'
 import { viemClientByNetworkId } from 'lib/viem-client'
 
 const client = viemClientByNetworkId[arbitrum.id]
 
-type UseTimeInPoolProps = {
+type UseTimeInPoolProps<SelectData = bigint> = {
   stakingAssetAccountAddress: string | undefined
+  select?: (timeInPoolSeconds: bigint) => SelectData
 }
 
+export const RFOX_CONTRACT_CREATION_BLOCK_NUMBER = 222913582n
```

Comment:
@woodenfurniture she back, we actually need this for the filters to work - made sure to use the creation time of the mainnet contract which was created a few hours before so this works both on the prod and testing contract

Reply:
Do we want to pull this with a query so we dont run into issues forgetting to update? 

Might be slow and unnecessary though, but here's a snippet

```typescript
    const [initLog] = await publicClient.getContractEvents({
      address: ARBITRUM_RFOX_PROXY_CONTRACT_ADDRESS,
      abi: stakingV1Abi,
      eventName: "Initialized",
      fromBlock: "earliest",
      toBlock: "latest",
    })
```

Reply:
We could absolutely do that, and while that would be the "proper" way it would be quite a heavy one indeed.
I'm thinking a more lightweight approache would be a simple mapping of contract to creation block number, i.e one for  the development contract, another for the prod one - if we're nice and exhaustive, this should be just as safe in theory

Reply:
Yes hardcode works considering the insane performance implications!

Reply:
dc299c223d084d808a5f8573474da4a426fab8cc